ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/PaginatingAsyncStream.scala

```diff
@@ -73,9 +72,7 @@ private[platform] object PaginatingAsyncStream {
           query(offset).map { result =>
             val nextPageOffset: Option[Off] = result.lastOption.map(getOffset)
             Some((nextPageOffset, result))
-          }(
-            DirectExecutionContext
-          ) // run in the same thread as the query, avoid context switch for a cheap operation
```

Comment:
> run in the same thread as the query

I would challenge this claim. I would also challenge any usage of `parasitic` as well, given that it's based on this assumption, which happens to be true most of the times, but not deterministically so. But that's definitely for another time.

Reply:
Agreed, which is why I deleted it. Someone else's problem, I am afraid.